### PR TITLE
fix: skip collection of deployments and replicaset with 0 replicas

### PR DESCRIPTION
**Pull request by mikenorgate**

### CODE REVIEW

File: internal/services/controller/controller.go [hunk 1]
```diff
@@ -687,20 +688,50 @@ type defaultInformer struct {
 	filters  filters.Filters
 }
 
-func getDefaultInformers(f informers.SharedInformerFactory) map[reflect.Type]defaultInformer {
+func getDefaultInformers(f informers.SharedInformerFactory, castwareNamespace string) map[reflect.Type]defaultInformer {
```
Comment (linkas45) [inline internal/services/controller/controller.go:691]: some tests would be nice

File: internal/services/controller/controller.go [hunk 2]
```diff
@@ -108,7 +108,7 @@ func CollectSingleSnapshot(ctx context.Context,
 	f := informers.NewSharedInformerFactory(clientset, 0)
 	df := dynamicinformer.NewDynamicSharedInformerFactory(dynamicClient, 0)
 
-	defaultInformers := getDefaultInformers(f)
+	defaultInformers := getDefaultInformers(f, "")
```
Comment (vladklokun) [inline internal/services/controller/controller.go:111]: question: why do we provide an empty string as a castware namespace here?

File: internal/services/controller/controller.go [hunk 3]
```diff
@@ -727,15 +758,6 @@ func createEventInformer(f informers.SharedInformerFactory, v version.Interface,
...
 
-func fetchAPIResourceLists(client discovery.DiscoveryInterface, log logrus.FieldLogger) []*metav1.APIResourceList {
-	_, apiResourceLists, err := client.ServerGroupsAndResources()
-	if err != nil {
-		log.Warnf("Error when getting server resources: %v", err.Error())
-		return nil
-	}
-	return apiResourceLists
-}
-
```
Comment (vladklokun) [inline internal/services/controller/controller.go:738]: question: what's the reason we deleted this function?

I remember seeing a linter complaining this unexported function was not used anywhere. Is this the reason?

Comment (mikenorgate) [reply internal/services/controller/controller.go:738]: Yeh, just cleanup of unused code

File: internal/services/controller/controller.go (hunk 2 above)
Comment (mikenorgate) [reply internal/services/controller/controller.go:111]: This is used in the offline snapshot dump, so we don't have a namespace, I'll move it to make it more obvious

File: internal/services/controller/controller.go (hunk 1 above)
Comment (mikenorgate) [reply internal/services/controller/controller.go:693]: added

**Review by vladklokun — APPROVED** [review]

**Review by linkas45 — APPROVED** [review]

**Pull request merged**